### PR TITLE
prepare research environment

### PR DESCRIPTION
**Pull request by V3RGANz**

### CODE REVIEW

File: data.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        if scaler is not None:
+            items = scaler.transform(items)
+
+        logging.info(f'input features: {list(data.columns)}')
+        logging.info(f'total number of input features: {len(data.columns)}')
+
+        return cls(items)
+
+
+def split_data(dataset, train_split, shuffle=False):
```
Comment (kazeevn) [inline data.py]: pls fix the seed

File: eval.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+def evaluate_model(generator, experiment, test_set, batch_size, batch_num, gan_noise_size, device, scaler):
+    predictions = []
+    for _ in tqdm(range(batch_num), desc='evaluation', position=0, leave=True):
+        x_noise = torch.randn((batch_size, gan_noise_size)).to(device)
+        predictions.append(generator(x_noise).cpu().detach().numpy())
+
+    predictions_np = np.concatenate(predictions)
+
+    inverse_generated = scaler.inverse_transform(predictions_np)
+    test_inverse = scaler.inverse_transform(test_set)
```
Comment (kazeevn) [inline eval.py]: Может быть, сразу передавать нетрансформированный test? Это даже логичнее с точки зрения названия `test_set`.

File: eval.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    inverse_generated = inverse_generated[:, [0, 1, 2, 3, 4, 6]]
```
Comment (kazeevn) [inline eval.py]: magic number 6 is not OK. Please at least make it a named constant.

File: init_scaler.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+data = pd.read_csv(infilename, delimiter=',', names=header)[features]
+data.dropna(inplace=True)
+X_train = data.values
+
+scaler = MinMaxScaler((-1, 1))
+scaler.fit(X_train)
+
+scaler_path = Path('GAN') / f'scaler.{level}.pkl'
+with scaler_path.open("wb") as file_scaler:
+    pickle.dump(scaler, file_scaler)
```
Comment (kazeevn) [inline init_scaler.py]: С точки зрения code style, лучше убрать выполяему код в main. https://www.journaldev.com/17752/python-main-function

**Review by kazeevn — CHANGES_REQUESTED** [review]

File: data.py (hunk 1 above)
Comment (V3RGANz) [reply data.py]: done

File: init_scaler.py (hunk 4 above)
Comment (V3RGANz) [reply init_scaler.py]: done

File: eval.py (hunk 2 above)
Comment (V3RGANz) [reply eval.py]: в принципе, было бы неплохо, если бы у нас train и test были 2 отдельных файла. я думаю, что можно сделать в отдельном модуле.

File: init_scaler.py (hunk 4 above)
Comment (V3RGANz) [reply init_scaler.py]: done

File: eval.py (hunk 3 above)
Comment (V3RGANz) [reply eval.py]: done

File: eval.py (hunk 2 above)
Comment (V3RGANz) [reply eval.py]: dataset splitting now located in `init_data.py`, also scaling back and forth not performing on test data

File: eval.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+        plt.xlim(min(bin_widths_t), max(bin_widths_t))
+
+        chi2 = stats.chisquare(count_t[start_bin[i]:], (count_g)[start_bin[i]:])
+        chisqs.append(chi2)
+
+    fig.show()
+
+    # FIXME: not working
+    # experiment.log_figure(figure_name='distributions', figure=fig)
```
Comment (V3RGANz) [inline eval.py]: I don't know why I get an error (warning) message 
```
COMET WARNING: Refuse to upload empty figure, please call log_figure before calling show 

COMET WARNING: Failing to save the matplotlib figure plt.subplots
```
I think it is because I'm running code in notebook/Colab

Comment (V3RGANz) [reply eval.py]: (in warning version I have no `fig.show` line)

File: init_scaler.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+def main():
+    header = [
+        "eventNumber", "weight",
+        "ljet1_pt", "ljet1_eta", "ljet1_phi", "ljet1_E", "ljet1_M",
+        "ljet2_pt", "ljet2_eta", "ljet2_phi", "ljet2_E", "ljet2_M",
+        "jj_pt", "jj_eta", "jj_phi", "jj_E", "jj_M",
+        "jj_dPhi", "jj_dEta", "jj_dR",
+    ]
+    features = [
```
Comment (kazeevn) [inline init_scaler.py]: `features` совпадает с `init_data.PTCL_FEATURES`. Одну и ту же сущность лучше не определять два раза, это создаёт потенциал для багов.

File: hyperparam.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
+from abc import abstractmethod
+from random import randint, uniform
+
+
+class RandArg:
```
Comment (kazeevn) [inline hyperparam.py:5]: Use `numpy.random`

File: init_data.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import pandas as pd
+
+from data import split_data, PTCL_HEADER, PTCL_FEATURES
+from util import fix_seed
+
+
+def main(args):
+    logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s', level=logging.INFO)
+
+    fix_seed(args.seed)
```
Comment (kazeevn) [inline init_data.py:16]: Move closer to split_data or use sklearn

File: init_scaler.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+from sklearn.preprocessing import MinMaxScaler
+
+from data import PTCL_HEADER, PTCL_FEATURES
+
+
+def main():
+    infilename = "csv/mg5_dijet_ht500.ptcl.pt250.nominal.csv"
+    level = infilename.split("/")[-1].split('.')[1]
+    data = pd.read_csv(infilename, delimiter=',', names=PTCL_HEADER)[PTCL_FEATURES]
+    data.dropna(inplace=True)
```
Comment (kazeevn) [inline init_scaler.py]: Проверьте, есть ли NA

File: init_scaler.py [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+import pandas as pd
+from sklearn.preprocessing import MinMaxScaler
+
+from data import PTCL_HEADER, PTCL_FEATURES
+
+
+def main():
+    infilename = "csv/mg5_dijet_ht500.ptcl.pt250.nominal.csv"
+    level = infilename.split("/")[-1].split('.')[1]
+    data = pd.read_csv(infilename, delimiter=',', names=PTCL_HEADER)[PTCL_FEATURES]
```
Comment (kazeevn) [inline init_scaler.py]: Эффективнее и идематичнее использовать `usecols`

File: main.py [hunk 11]
```diff
@@ -0,0 +1,82 @@
+import argparse
+import logging
+from datetime import datetime
+from pathlib import Path
+
+from comet_ml import Experiment
+import torch
+
+from data import get_data
+from eval import evaluate_model
```
Comment (kazeevn) [inline main.py]: `eval` - зарезервированное слово в Питоне, не надо использовать его для названия модуля

File: main.py [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+    experiment = Experiment('gflIAsawYkIJvtkFb55lOwno7', project_name="sirius-gan-tails", workspace="v3rganz")
+    experiment.log_parameters(vars(args))
+
+    epochs_trained = train(generator, discriminator, args, dataset_train, optimizer_g, optimizer_d, scaler=scaler,
+                           save_dir=save_dir, test_dataset=dataset_test.items[:len(dataset_test) // 10],
+                           ecaluate_every=args.log_every, experiment=experiment, device=device)
+
+    n_events = len(dataset_test)
+    steps = n_events // 512
```
Comment (kazeevn) [inline main.py]: Please make `512` from magic number into a named constant

File: model.py [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+            Flatten(),
+            nn.Linear(1024, gan_output_size),
+            nn.Tanh()
+        )
+
+    def forward(self, x):
+        return self.layers(x)
+
+
+class GeneratorFC(nn.Module):
```
Comment (kazeevn) [inline model.py:46]: Please make the number of parameters equal to CNN

**Review by kazeevn — CHANGES_REQUESTED** [review]

File: init_scaler.py (hunk 6 above)
Comment (V3RGANz) [reply init_scaler.py]: ты посмотрел старую версию, это уже исправлено)

File: hyperparam.py (hunk 7 above)
Comment (V3RGANz) [reply hyperparam.py:5]: why `numpy`?

File: init_scaler.py (hunk 9 above)
Comment (V3RGANz) [reply init_scaler.py]: их нет

File: model.py (hunk 13 above)
Comment (V3RGANz) [reply model.py:46]: done

File: main.py (hunk 12 above)
Comment (V3RGANz) [reply main.py]: made parameter

File: main.py (hunk 11 above)
Comment (V3RGANz) [reply main.py]: renamed

File: init_scaler.py (hunk 10 above)
Comment (V3RGANz) [reply init_scaler.py]: done

File: init_data.py (hunk 8 above)
Comment (V3RGANz) [reply init_data.py:16]: using sklearn

**Review by kazeevn — APPROVED** [review]
LGTM
